### PR TITLE
#8572 - Updated docs to include pip >= 20.3b1 requirement

### LINKED ISSUE #8572: pip_package_info: "no such option: --format" (--format added in pip 20.3b1)
### Summary

Listing pip packages, the task returns:

```
[WARNING]: Failed to retrieve packages with /some/path/to/some/pip: Unable to list
packages rc=2 :  Usage:      pip list [options]  no such option: --format
```

The reason is that pip only added the `--format` option in version 20.3b1 (see https://pip.pypa.io/en/stable/news/#b1-2020-10-31)

So either lower pip versions have to omit the `--format` option and extract package info otherwise OR **you should add the minimum required pip version to the pip_package_info docs** (see https://docs.ansible.com/ansible/latest/collections/community/general/pip_package_info_module.html#requirements)

Additional info: The used pip version was 8.1.1 so really old. pip_package_info should have been used to detect its version and then update that to a newer one.

### Issue Type

Documentation Report

### Component Name

pip_package_info

### Ansible Version

```console (paste below)
(ansible-8.7.0) vagrant@ansible-cm:~/playbooks/playbook-mahara$ ansible --version
ansible [core 2.15.8]
  config file = /home/vagrant/playbooks/playbook-mahara/ansible.cfg
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/vagrant/python-venvs/ansible-8.7.0/lib/python3.10/site-packages/ansible
  ansible collection location = /home/vagrant/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/vagrant/python-venvs/ansible-8.7.0/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/home/vagrant/python-venvs/ansible-8.7.0/bin/python)
  jinja version = 3.0.3
  libyaml = True
```

### Community.general Version

```console (paste below)
# /home/vagrant/python-venvs/ansible-8.7.0/lib/python3.10/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 7.5.2
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

_No response_

### Additional Information

If the docs are updated, users better understand when this module can be used.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by danowar2k**
##### SUMMARY

Fixes #8572 (include pip >= 20.3b1 requirement in docs)

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
pip_package_info

##### ADDITIONAL INFORMATION
See #8572 for the problem

### CODE REVIEW

File: plugins/modules/pip_package_info.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
     type: list
     elements: path
 requirements:
+  - pip >= 20.3b1 (necessary for the --format option)
```
Comment (felixfontein) [inline plugins/modules/pip_package_info.py:30]:
```suggestion
  - pip >= 20.3b1 (necessary for the C(--format) option)
```

Comment (danowar2k) [reply plugins/modules/pip_package_info.py:30]: Sorry, I don't know the formatting commands from the docs. If that means "cursive", I approve :-)

Comment (danowar2k) [reply plugins/modules/pip_package_info.py:30]: Ah, https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html . Monospace. Well then :-)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@danowar2k thanks a lot for improving the docs!

**Pull request merged**